api/app/jobs/common_model_fetchers.py

```diff
@@ -304,6 +304,32 @@ def transform(long, lat):
         # Add this prediction to the session (we'll commit it later.)
         self.session.add(station_prediction)
 
+    def _calculate_past_24_hour_precip(self, station: WeatherStation, model_run: PredictionModelRunTimestamp,
+                                       prediction: ModelRunGridSubsetPrediction, station_prediction: WeatherStationModelPrediction):
+        """ Calculate the predicted precipitation over the previous 24 hours within the specified model run.
+        If the model run does not contain a prediction timestamp for 24 hours prior to the current prediction,
+        return the predicted precipitation from the previous run of the same model for the same time frame. """
+        start_prediction_timestamp = prediction.prediction_timestamp - timedelta(days=1)
+        # Check if a prediction exists for this model run 24 hours in the past
+        previous_prediction_from_same_model_run = get_weather_station_model_prediction(self.session, station.code,
+                                                                                       model_run.id, start_prediction_timestamp)
+        # If a prediction from 24 hours ago from the same model run exists, return the difference in cumulative precipitation
```

Comment:
Does this assume a run happens every 24 hours for a model? Should we do some timestamp checks to make sure? What if a job fails to run and we get a previous timestamp from several days before?

Reply:
No, it doesn't make any assumptions about the frequency at which a model runs. It might be easier to have a dev chat about this?

Reply:
Within a given model run, it is not possible to get cumulative precipitation for the previous 24 hour period during the first 23 hours the model is making predictions for, we would end up missing some hours of potential accumulation. Instead, I go to the most recent prior model run and get the 24 hour precip generated by that numerical weather model for the same time period. The data ends up being slightly stale as it is from a previous model run, but it is the best we can do.